### PR TITLE
Web voice assistant

### PR DESCRIPTION
**Pull request by ksyeo1010**

### CODE REVIEW

**Review by ErisMik — APPROVED** [review]

File: recipes/llm-voice-assistant/web/public/index.html [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      float: right;
+
+      &:hover {
+        color: #377dff;
+      }
+    }
+  </style>
+</head>
+<body>
+  <h1>picoLLM Web Voice Assistant</h1>
```
Comment (laves) [inline recipes/llm-voice-assistant/web/public/index.html:124]:
```suggestion
  <h1>LLM Voice Assistant</h1>
```

File: recipes/llm-voice-assistant/web/public/controller.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        onDetection: () => {
+          message.innerText = "Wake word detected, utter your request or question...";
+          humanElem = startHumanMessage();
+        },
+        onTranscript: (transcript) => {
+          completeTranscript += transcript;
+          addMessage(humanElem, transcript);
+        },
+        onEndpoint: async () => {
+          message.innerText = "Generating...";
```
Comment (laves) [inline recipes/llm-voice-assistant/web/public/controller.js:115]: can this be the moving `...`? Or add a progress spinner?

File: recipes/llm-voice-assistant/web/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    '<|eot_id|>', // Llama-3
+  ];
+
+  const onCheetahFlushed = async (): Promise<void> => {
+    const prompt = transcripts.join('');
+    transcripts = [];
+    dialog.addHumanRequest(prompt);
+
+    const { completion, completionTokens } = await pllm.generate(dialog.prompt(), {
+      completionTokenLimit: 128,
```
Comment (laves) [inline recipes/llm-voice-assistant/web/src/main.ts:113]: you could include stop phrases, since we have a list of those

**Review by laves — CHANGES_REQUESTED** [review]

**Comment by laves** [comment]
more margine between the bubbles

**Review by laves — APPROVED** [review]

**Pull request merged**